pkg/reconciler/revision/resources/deploy.go

```diff
@@ -141,6 +141,23 @@ func makePodSpec(rev *v1.Revision, loggingConfig *logging.Config, tracingConfig
 		return nil, fmt.Errorf("failed to create queue-proxy container: %w", err)
 	}
 
+	userContainer := BuildUserContainer(rev)
+	podSpec := BuildPodSpec(rev, []corev1.Container{*userContainer, *queueContainer})
+
+	// Add the Knative internal volume only if /var/log collection is enabled
+	if observabilityConfig.EnableVarLogCollection {
+		podSpec.Volumes = append(podSpec.Volumes, internalVolume)
+	}
+
+	if autoscalerConfig.EnableGracefulScaledown {
+		podSpec.Volumes = append(podSpec.Volumes, labelVolume)
+	}
+
+	return podSpec, nil
+}
+
+// BuildUserContainer makes a container from the Revision template.
```

Comment:
Do those have to be public methods?

Reply:
They do!

The upcoming CL uses this shared logic to make a user container and validate it against k8s

Reply:
Both or just `BuildUserContainer`?

Reply:
https://gist.github.com/whaught/c847766071059dc7ebb9077f3fb974e8

It's possible to combine them, but I'm not sure if it'd be worth it.